piker/data/__init__.py

```diff
@@ -112,24 +97,25 @@ async def maybe_spawn_brokerd(
             yield portal
 
         else:  # no daemon has been spawned yet
+            async with maybe_spawn_pikerd(
+                loglevel=loglevel
+            ) as pikerd_portal:
+
+                if pikerd_portal is None:
+                    await spawn_brokerd(
```

Comment:
Hmm this is interesting. I wonder if we can avoid having to check for `None` here (because we're likely going to see this pattern often)  and instead hand some kind of *local portal* thing kinda like [what's already inside `tractor`](https://github.com/goodboy/tractor/blob/master/tractor/_portal.py#L374) - but maybe it's over the top?

it would be nice to have the polymorphism to avoid all this branch logic.